src/buffer/out/textBuffer.cpp

```diff
@@ -803,31 +883,23 @@ void TextBuffer::ScrollRows(const til::CoordType firstRow, const til::CoordType
         // | 2
         // | 3
         // | 4
-        // | 5 A. begin + firstRow
+        // | 5 A. firstRow
         // | 6
         // | 7
-        // | 8 B. begin + firstRow + size
+        // | 8 B. firstRow + size
         // | 9
-        // | 10 C. begin + firstRow + size + delta
+        // | 10 C. firstRow + size + delta
         // | 11
         // - end
         // We want B-1 to slide down to C-1 (the positive delta) and everything from [A, B) to slide down with it.
-        // So the final layout will be
-        // --- (storage) ----
-        // | 0 begin
-        // | 1
-        // | 2
-        // | 3
-        // | 4
-        // | 8
-        // | 9
-        // | 5
-        // | 6
-        // | 7
-        // | 10
-        // | 11
-        // - end
-        std::rotate(_storage.begin() + firstRow, _storage.begin() + firstRow + size, _storage.begin() + firstRow + size + delta);
+        y = firstRow + size - 1;
+        end = firstRow - 1;
+        step = -1;
+    }
+
+    for (; y != end; y += step)

```

Comment:
I don't completely understand why this doesn't have the same problem as memcpy. Depending on the direction you are moving, we need to copy end-to-start or start-to-end, right?

Reply:
oh, i see. we _are_. this is just a generic loop that goes both ways. Oi.

Reply:
If you prefer we can make it 2 separate loops! I initially had 2 loops, and wrote this single loop because I was curious if I could.